eth/client.go

```diff
@@ -431,24 +431,32 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
 		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
 	}
 
-	// Get current delegate
+	// Get delegator
 	delegator, err := c.GetDelegator(sender)
 	if err != nil {
 		return nil, errors.Wrapf(err, "unable to get delegator")
 	}
 
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
 	// Switching delegate's calculate old delegate positions
 	var oldHints lpTypes.TranscoderPoolHints
 	if delegator.DelegateAddress != to {
-		oldHints = findTranscoderHints(delegator.DelegateAddress, transcoders)
+		// Get total bonded
+		totalBonded, err := c.TranscoderTotalStake(delegator.DelegateAddress)
+		if err != nil {
+			return nil, err
+		}
+		oldHints = simulateTranscoderPoolUpdate(delegator.DelegateAddress, new(big.Int).Sub(totalBonded, delegator.BondedAmount), transcoders, isFull)
```

Comment:
`delegator.BondedAmount` needs to be replaced with a `PendingStake()` call because `delegator.BondedAmount` will not include accumulated rewards until after the those rewards are automatically claimed & credited to `delegator.BondedAmount` at the start of the execution of the bond tx.